tests/unit/components/menu/GitBranchMenu.spec.js

```diff
@@ -0,0 +1,153 @@
+import { useRoute } from 'vue-router';
+import { getBranches } from 'src/composables/Project';
+import Branch from 'src/models/git/Branch';
+import { createI18n } from 'vue-i18n';
+import i18nConfiguration from 'src/i18n';
+import { installQuasarPlugin } from '@quasar/quasar-app-extension-testing-unit-jest';
+import GitEvent from 'src/composables/events/GitEvent';
+
+installQuasarPlugin();
+
+jest.mock('vue-router', () => ({
+  useRoute: jest.fn(),
+}));
+
+jest.mock('src/composables/events/GitEvent', () => ({
+  FetchEvent: {
+    subscribe: jest.fn(),
+  },
+}));
+
+jest.mock('src/composables/Project', () => ({
+  getCurrentBranch: jest.fn(() => Promise.resolve('main')),
+  getBranches: jest.fn(() => Promise.resolve([])),
+  fetchGit: jest.fn(() => Promise.resolve()),
+}));
+
+describe('Test component: GitBranchMenu', () => {
+  let wrapper;
+  let subscribe;
+  let unsubscribe;
+
+  useRoute.mockImplementation(() => ({
+    params: {
+      projectName: 'project-00000000',
+      viewType: 'model',
+    },
+  }));
+
+  beforeEach(() => {
+    subscribe = jest.fn();
+    unsubscribe = jest.fn();
+    GitEvent.FetchEvent.subscribe.mockImplementation(() => {
+      subscribe();
+      return { unsubscribe };
+    });
+    wrapper = shallowMount(GitBranchMenu, {
+      props: {
+        currentBranchName: 'main',
+      },
+      mocks: {
+        DialogEvent,
+      },
+      global: {
+        plugins: [
+          createI18n(i18nConfiguration),
+        ],
+      },
+    });
+  });
+
+  describe('Test functions', () => {
+    describe('Test function: openCloseExpandMenu', () => {
+      it('Should invert local value', () => {
+        expect(wrapper.vm.showLocal).toEqual(false);
+        expect(wrapper.vm.showRemote).toEqual(false);
+
+        wrapper.vm.openCloseExpandMenu(true);
+        expect(wrapper.vm.showLocal).toEqual(true);
+        expect(wrapper.vm.showRemote).toEqual(false);
+
+        wrapper.vm.openCloseExpandMenu(true);
+        expect(wrapper.vm.showLocal).toEqual(false);
+        expect(wrapper.vm.showRemote).toEqual(false);
+      });
+
+      it('Should invert remote value', () => {
+        expect(wrapper.vm.showLocal).toEqual(false);
+        expect(wrapper.vm.showRemote).toEqual(false);
+
+        wrapper.vm.openCloseExpandMenu(false);
+        expect(wrapper.vm.showLocal).toEqual(false);
+        expect(wrapper.vm.showRemote).toEqual(true);
+
+        wrapper.vm.openCloseExpandMenu(false);
+        expect(wrapper.vm.showLocal).toEqual(false);
+        expect(wrapper.vm.showRemote).toEqual(false);
+      });
+    });
+
+    describe('Test function: initBranches', () => {
+      it('Should be empty on no branches', async () => {
+        getBranches.mockImplementation(() => Promise.resolve([]));
+        await wrapper.vm.initBranches();
+        expect(wrapper.vm.branches).toEqual({ local: [], remote: [] });
+      });
+
+      it('Local branch array should contains all local branches', async () => {
```

Comment:
`Local branch array should contains all local branches` => `should fill local branch array with all local branches`

Reply:
done